test/gtest/conv_igemm_dynamic_xdlops.cpp

```diff
@@ -0,0 +1,224 @@
+        tokens.push_back(*begin++);
+}
+
+class Conv2dFloat : public testing::TestWithParam<std::vector<TestCase>>
+{
+};
+
+class Conv2dHalf : public testing::TestWithParam<std::vector<TestCase>>
+{
+};
+
+void Run2dDriver(miopenDataType_t prec)
+{
+
+    std::vector<TestCase> params;
+    switch(prec)
+    {
+    case miopenFloat: params = Conv2dFloat::GetParam(); break;
+    case miopenHalf: params = Conv2dHalf::GetParam(); break;
+    case miopenInt8:
+    case miopenBFloat16:
+    case miopenInt8x4:
+    case miopenInt32:
+    case miopenDouble:
+    case miopenFloat8:
+    case miopenBFloat8:
+        FAIL() << "miopenInt8, miopenBFloat16, miopenInt8x4, miopenInt32, "
+                  "miopenDouble, miopenFloat8, miopenBFloat8 "
+                  "data type not supported by conv_igemm_dynamic_xdlops test";
+
+    default: params = Conv2dFloat::GetParam();
+    }
+
+    for(const auto& test_value : params)
+    {
+        std::vector<std::string> tokens;
+        GetArgs(test_value, tokens);
+        std::vector<const char*> ptrs;
+
+        std::transform(tokens.begin(),
+                       tokens.end(),
+                       std::back_inserter(ptrs),
+                       [](const std::string& str) { return str.data(); });
+
+        testing::internal::CaptureStderr();
+        test_drive<conv2d_driver>(ptrs.size(), ptrs.data());
+        auto capture = testing::internal::GetCapturedStderr();
+        std::cout << capture;
+    }
+};
+
+bool IsTestSupportedForDevice(const miopen::Handle& handle)
+{
+    std::string devName = handle.GetDeviceName();
+    if(devName == "gfx908" || miopen::StartsWith(devName, "gfx94"))
+        return true;
+    else
+        return false;
+}
+
+TEST_P(Conv2dFloat, FloatTest)
+{
+    const auto& handle = get_handle();
+    if(IsTestSupportedForDevice(handle) && !SkipTest())
+    {
+        Run2dDriver(miopenFloat);
+    }
+    else
+    {
+        GTEST_SKIP();
+    }
+};
+
+TEST_P(Conv2dHalf, HalfTest)
+{
+    const auto& handle = get_handle();
+    if(IsTestSupportedForDevice(handle) && !SkipTest())
+    {
+        Run2dDriver(miopenHalf);
+    }
+    else
+    {
+        GTEST_SKIP();
+    }
+};
+
+std::vector<TestCase> GetTestCases(const std::string& precision)
+{
+
+    std::vector<std::string> env_bwd_xdlops = {
+        "MIOPEN_FIND_MODE=normal",
+        "MIOPEN_DEBUG_FIND_ONLY_SOLVER=ConvAsmImplicitGemmGTCDynamicBwdXdlops"};
+
+    std::vector<std::string> env_fwd_xdlops = {
+        "MIOPEN_FIND_MODE=normal",
+        "MIOPEN_DEBUG_FIND_ONLY_SOLVER=ConvAsmImplicitGemmGTCDynamicFwdXdlops"};
+
+    std::vector<std::string> env_wrw_xdlops = {
+        "MIOPEN_FIND_MODE=normal",
+        "MIOPEN_DEBUG_FIND_ONLY_SOLVER=ConvAsmImplicitGemmGTCDynamicWrwXdlops"};
+
+    std::string v           = " --verbose";
+    std::string dis_bk_data = " --disable-backward-data";
+    std::string dis_bk_wei  = " --disable-backward-weights";
+    std::string dis_fwd     = " --disable-forward";
```

Comment:
Can we make it `std::string_view` or at least const?

Reply:
> Can we make it `std::string_view` or at least `const`?

@CAHEK7 , `std::string_view` or at least `const` is a good idea :)

Reply:
> Can we make it `std::string_view` or at least `const`?

std::string_view would give an error. Updated with const

/MIOpen/test/gtest/conv_igemm_dynamic_xdlops.cpp:180:40: error: invalid operands to binary expression ('const std::string' (aka 'const basic_string<char>') and 'std::string_view' (aka 'basic_string_view<char>'))
    TestCase{env_bwd_xdlops, precision + v + " --input  400  256 1 1 --weights 1024  256  1 1 --pads_strides_dilations 0 0 1 1 1 1" + dis_fwd + dis_bk_wei},

Reply:
Yes, `std::string_view` is immutable and does not support altering operations.
And it seems `std::string` only supports '.append' method to add `stding_view`s.
So either it should be rewritten to adapt this behavior or we can use `const std::string`.

Reply:
> So either it should be rewritten to adapt this behavior or we can use `const std::string`.

@CAHEK7 , I had updated with const std::string.